src/metadata.py

```diff
@@ -62,3 +93,55 @@ def get(path: Path) -> types_.Metadata:
         raise InputError(f"Invalid value for docs key: {docs}, expected a string value")
 
     return types_.Metadata(name=name, docs=docs)
+
+
+def _parse_charmcraft_yaml(charmcraft_yaml: Path) -> types_.Metadata:
+    """Parse charmcraft file.
+
+    Args:
+        charmcraft_yaml: The file path the the charmcraft file.
+
+    Returns:
+        The contents of the charmcraft file.
+
+    Raises:
+        InputError: if the charmcraft file does not exist or are malformed.
+    """
+    try:
+        charmcraft = yaml.safe_load(charmcraft_yaml.read_text())
```

Comment:
Minor: Could a FileNotFoundError happen here?

Reply:
in the get function that calls this one, the file is checked, so it should exist by this point.